### ISSUE TITLE
Incorrect datatype in payload examples

### ISSUE THREAD

**Issue by stmccartneyncc** [eclipse-sparkplug/sparkplug#185]
I think the payload examples in [Section 6 - Payloads](https://github.com/eclipse/sparkplug/blob/develop/specification/src/main/asciidoc/chapters/Sparkplug_6_Payloads.adoc) are incorrect.  They currently (for example) look like this:
```
{
  "name": "bdSeq",
  "timestamp": 1486144502122,
  "dataType": "Uint64",
  "value": 0
}
```
and I think they should be:
```
{
  "name": "bdSeq",
  "timestamp": 1486144502122,
  "dataType": 8,
  "value": 0
}
```
Is that right?  I'm happy to go and make the changes and submit a pull request, just wanted to get agreement first that it's correct.

**Comment by jbrzozoski** [comment]
The datatypes are officially enumerated as part of the protobuf definition now. I'd argue that it would make more sense to use those and have something like:

```
{
  "name": "bdSeq",
  "timestamp": 1486144502122,
  "dataType": DataType.UInt64,
  "value": 0
}
```

**Comment by stmccartneyncc** [comment]
Good point, I hadn't spotted that change.  My comment is based on this entry from the datatype definition under Metric:
`The datatype MUST be an unsigned 32-bit integer representing the datatype`
However I can see now that it also says this, which is ambiguous:
`The datatype MUST be one of the enumerated values as shown in the valid Sparkplug Data Types`

There's not a lot of point having the DataType enum defined if it's never used(?), so I would suggest the two options are:
1. Update the datatype text to specify that the enum reference should be used and not an integer, change the protobuf spec so the type of datatype is enum (which will break backwards compatibility) and change the examples accordingly.
2. Go back to the previous approach where the datatype definitions were commented out of the protobuf definition, and update the examples as per my original comment.

I'm only just starting to use Sparkplugs, so don't really feel I have the background to make any kind of recommendation.  Option 1 feels like the 'right' thing to do, but I assume that breaking backwards compatibility is a no-no?

**Comment by I-Campbell** [comment]
I think the backward compatibility is only that the code which is auto-generated by the Google protobuf thing will change. Actually the bytes on the wire would not change.

> {
>   "name": "bdSeq",
>   "timestamp": 1486144502122,
>   "dataType": DataType.UInt64,
>   "value": 0
> }
This is not valid JSON, it would need to be with quotes around "DataType.UInt64".
There is no official sparkplug JSON encoding, it is only to give a human readable idea of what the spec is trying to convey.
It is tricky that we want to convey UInt64, but the payload will actually be an integer type.

Maybe instead of JSON, we could use hjson, in which we can convey things better.

**Comment by wes-johnson** [comment]
I did modify the proto definition to use the DataType enumeration: https://github.com/eclipse-sparkplug/sparkplug/pull/550. However, I'm not going to change the JSON examples in the spec. It seems readable as is to me and will now be similar to the protoc encode/decode text format. Ultimately, this will also match the JSON that is recommended per https://github.com/eclipse-sparkplug/sparkplug/issues/231.
